syncstorage/db.go

```diff
@@ -922,22 +928,11 @@ func (d *DB) getBSOs(
 
 	var more bool
 	var nextOffset int
-
-	// if a limit was applied, determine if there are records beyond
-	// the upper bound and return the pertinent information
-	if limit > 0 {
-		var totalRows int
-		countQuery := "SELECT COUNT(1) NumRows FROM BSO " + where + " " + orderBy
-		if err := tx.QueryRow(countQuery, values...).Scan(&totalRows); err != nil {
-			return nil, err
-		}
-
-		if totalRows > limit+offset {
-			more = true
-			nextOffset = offset + limit
-		}
-	} else {
-		more = false
+	num := len(bsos)
+	if limit > 0 && num > limit {
+		bsos = bsos[:num-1]
```

Comment:
ditto here re: handling of `limit=0`, which nobody should ever use, but which we should probably not break.  Perhaps we need explicit functional tests cases for this case?